File: VB/GrowthWare.WebApplication/Controllers/SessionStateRouteHandler .vb
Imports System.Web.Routing
Imports System.Web.SessionState
Imports System.Web.Http.WebHost

Public Class SessionStateRouteHandler
    Implements IRouteHandler

    Public Function GetHttpHandler(requestContext As RequestContext) As IHttpHandler Implements IRouteHandler.GetHttpHandler
        Return New SessionableControllerHandler(requestContext.RouteData)
    End Function
End Class

File: Net_Framework/CS/GrowthWare.WebApplication/Controllers/SessionStateRouteHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Routing;

namespace GrowthWare.WebApplication.Controllers
{
    public sealed class SessionStateRouteHandler : IRouteHandler
    {
        IHttpHandler IRouteHandler.GetHttpHandler(RequestContext requestContext)
        {
            return new SessionableControllerHandler(requestContext.RouteData);
        }
    }
}